### PR TITLE
Map configuration

### PR DESCRIPTION
**Pull request by tonio**
Adds:
 - position
 - positionSrs
 - zoom
 - bgLayer
 - layers
 - add `addLayerById` to `lux.Map` object.

### CODE REVIEW

**Comment by tonio** [comment]
I’ve added a fake response for the layer list.

File: jsapi/externs/luxx.js [hunk 1]
```diff
@@ -3,6 +3,59 @@
...
 var luxx;
 
+/**
+ * Object literal with config options for the map.
+ * @typedef {{
+ *     bgLayer: (string),
+ *     layers: (Array<string>),
+ *     target: (Element|string),
+ *     position: (ol.Coordinate|undefined),
+ *     positionSrs: (string|undefined),
```
Comment (pgiraud) [inline jsapi/externs/luxx.js]: I think it would be safier if we also accept `number` for `positionSrs` and `layers`.

File: jsapi/externs/luxx.js [hunk 2]
```diff
@@ -3,6 +3,59 @@
...
+ * @type {string}
+ * @api stable
+ */
+luxx.MapOptions.prototype.positionSrs;
+
+/**
+ * Zoom level used to calculate the initial resolution for the view.
+ * @type {number|undefined}
+ */
+luxx.MapOptions.prototype.zoom;
```
Comment (pgiraud) [inline jsapi/externs/luxx.js]: Any reason why `target` and `positionSrs` are api stable and no the other properties?

File: jsapi/src/main.js [hunk 3]
```diff
@@ -6,28 +6,75 @@ goog.require('ol.events');
...
+  var layers    = [];
+  var defaultBg = 'basemap_2015_global';
+
+  /**
+   * @private
+   * @type {Object} config A JSON layers config object.
+   */
+  this.layersConfig = null;
+
+  this.promise = null;
```
Comment (pgiraud) [inline jsapi/src/main.js]: I'm not sure if it is required or not but `promise` could be documented as `@private` as well.

File: jsapi/externs/luxx.js [hunk 4]
```diff
@@ -3,6 +3,59 @@
...
+ * @api stable
+ */
+luxx.MapOptions.prototype.target;
+
+/**
+ * The initial center for the map view. The coordinate system for the center is
+ * specified with the `positionSrs` option.
+ * @type {ol.Coordinate|undefined}
+ */
+luxx.MapOptions.prototype.center;
```
Comment (pgiraud) [inline jsapi/externs/luxx.js]: Did you mean `position`?

File: jsapi/src/main.js [hunk 5]
```diff
@@ -6,28 +6,75 @@ goog.require('ol.events');
...
+  }
+  this.promise = fetch('../layers.json').then(function(resp) {
+    return resp.json();
+  }).then(function(json) {
+    this.layersConfig = /** @type {luxx.LayersOptions} */ (json);
+    this.addLayers_(layers);
+  }.bind(this));
+
+  var viewOptions = {
+    center : ol.proj.fromLonLat([6.215, 49.845]),
```
Comment (pgiraud) [inline jsapi/src/main.js]: What's the purpose of `fromLonLat` here? Didn't you forget to convert this to `3857`?

Comment (tonio) [inline jsapi/src/main.js]: `fromLonLat` converts by default to `3857`.

File: jsapi/externs/luxx.js (hunk 4 above)
Comment (tonio) [inline jsapi/externs/luxx.js]: yep

File: jsapi/src/main.js (hunk 3 above)
Comment (tonio) [inline jsapi/src/main.js]: Not required, but will do.

File: jsapi/externs/luxx.js (hunk 2 above)
Comment (tonio) [inline jsapi/externs/luxx.js]: No, I’ll fix it.

File: jsapi/externs/luxx.js (hunk 1 above)
Comment (tonio) [inline jsapi/externs/luxx.js]: Sure.

**Comment by pgiraud** [comment]
I'd rather squash the commits into one.
Otherwise looks good. Please merge.

**Pull request merged**